File: sokrates/src/main/java/de/vorb/sokrates/properties/IndexProperties.java
package de.vorb.sokrates.properties;

import lombok.Data;
import org.springframework.stereotype.Component;

import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;
import java.util.Locale;
import java.util.Properties;

@Data
@Component
public class IndexProperties {

    private String name;
    private String title;
    private Locale locale;

    private String template;
    private Path outputFile;

    private List<String> orderBy = new ArrayList<>();
    private String where;
    private Integer limit;

    private Grouping grouping = Grouping.NONE;

    private Properties properties = new Properties();

}

File: src/main/kotlin/de/vorb/sokrates/config/IndexProperties.kt
package de.vorb.sokrates.config

import java.util.*

class IndexProperties {

    var name: String = ""

    var itemType: String = ""

    var outputPattern: String = ""

    var template: String = ""

    var itemsPerPage: Long = Long.MAX_VALUE

    var maxItems: Long = Long.MAX_VALUE

    val groupBy: List<String> = ArrayList()

    val orderBy: List<OrderProperties> = ArrayList()

    val misc: Map<String, String> = HashMap()

}
